AzureMonitorAgent/agent.py

```diff
@@ -204,14 +208,75 @@ def install():
 
-    default_configs = {
-        "ENABLE_MCS" : "true",
-        "MCS_ENDPOINT" : "eastus.amcs.ppe.control.monitor.azure.com",
+    default_configs = {        
+        "MCS_ENDPOINT" : "amcs.control.monitor.azure.com",
         "AZURE_ENDPOINT" : "https://management.azure.com/",
-        "ADD_REGION_TO_MCS_ENDPOINT" : "false",
-        "OMS_TLD" : "int2.microsoftatlanta-int.com",
-        "customResourceId" : "/subscriptions/42e7aed6-f510-46a2-8597-a5fe2e15478b/resourcegroups/amcs-test/providers/Microsoft.OperationalInsights/workspaces/amcs-pretend-linuxVM"
+        "ADD_REGION_TO_MCS_ENDPOINT" : "true",
+        #"OMS_TLD" : "int2.microsoftatlanta-int.com",
+        #"customResourceId" : "/subscriptions/42e7aed6-f510-46a2-8597-a5fe2e15478b/resourcegroups/amcs-test/providers/Microsoft.OperationalInsights/workspaces/amcs-pretend-linuxVM",
+        #"ENABLE_MCS" : "true",
     }
+
+    # decide the mode
+    if protected_settings is None:
+        default_configs["ENABLE_MCS"] = "true"
+    else:
+        # look for LA protected settings
+        for var in protected_settings.keys():
+            if "_key" in var:
+                default_configs[var] = protected_settings.get(var)
+        
+        # check if required GCS params are available
+        MONITORING_GCS_CERT_CERTFILE = None
+        if protected_settings.has_key("certificate"):
+            MONITORING_GCS_CERT_CERTFILE = base64.standard_b64decode(protected_settings.get("certificate"))
+
+        MONITORING_GCS_CERT_KEYFILE = None
+        if protected_settings.has_key("certificateKey"):
+            MONITORING_GCS_CERT_KEYFILE = base64.standard_b64decode(protected_settings.get("certificateKey"))
+
+        MONITORING_GCS_ENVIRONMENT = ""
+        if protected_settings.has_key("monitoringGCSEnvironment"):
+            MONITORING_GCS_ENVIRONMENT = protected_settings.get("monitoringGCSEnvironment")
+
+        MONITORING_GCS_NAMESPACE = ""
+        if protected_settings.has_key("namespace"):
+            MONITORING_GCS_NAMESPACE = protected_settings.get("namespace")
+
+        MONITORING_GCS_ACCOUNT = ""
+        if protected_settings.has_key("monitoringGCSAccount"):
+            MONITORING_GCS_ACCOUNT = protected_settings.get("monitoringGCSAccount")
+
+        MONITORING_GCS_REGION = ""
+        if protected_settings.has_key("monitoringGCSRegion"):
+            MONITORING_GCS_REGION = protected_settings.get("monitoringGCSRegion")
+
+        MONITORING_CONFIG_VERSION = ""
+        if protected_settings.has_key("configVersion"):
+            MONITORING_CONFIG_VERSION = protected_settings.get("configVersion")
+
+        if MONITORING_GCS_CERT_CERTFILE is None or MONITORING_GCS_CERT_KEYFILE is None or MONITORING_GCS_ENVIRONMENT is "" or MONITORING_GCS_NAMESPACE is "" or MONITORING_GCS_ACCOUNT is "" or MONITORING_GCS_REGION is "" or MONITORING_CONFIG_VERSION is "":
+            default_configs["ENABLE_MCS"] = "true"
```

Comment:
Why not having an explicit switch to enable MCS? A misconifg in GCS does not imply MCS is enabled.

Reply:
MCS is enabled by default, but I now removed the code to enable it when GCS is misconfigured